odl/test/tomo/geometry/geometry_test.py

```diff
@@ -549,6 +550,48 @@ def test_fanbeam_frommatrix():
                                                    det_rad, sing_mat)
 
 
+def test_fanbeam_flying_focal_spot():
+    """Test the flying focal spot in 2d fan beam geometry."""
+    full_angle = np.pi
+    apart = odl.uniform_partition(0, full_angle, 10)
+    dpart = odl.uniform_partition(0, 1, 10)
+    src_rad = 10
+    det_rad = 5
+    # Source positions with flying focal spot should correspond to
+    # source positions of 2 geometries with different starting positions
+    shift1 = np.array([2, -3])
+    shift2 = np.array([-2, 3])
+    init = np.array([1, 0], dtype=np.float32)
+    geom_ffs = odl.tomo.FanBeamGeometry(
+        apart, dpart,
+        src_rad, det_rad,
+        src_to_det_init=init,
+        src_shift_func=lambda angle: flying_focal_spot(
```

Comment:
This lambda is rather big, prefer creating a function outside or using functools.partial